javascript/ql/test/query-tests/Security/CWE-079/DomBasedXssWithResponseThreat/testReactRelay.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import { useFragment } from 'react-relay';
+
+const func1 = ({ commentRef, query }) => {
+  const commentData = useFragment(query, commentRef); // $ Source=[js/xss]
+  return (
+    <p dangerouslySetInnerHTML={{ __html: commentData.text }}> // $ Alert=[js/xss]
+      {" "}
+      {commentData.text}
+    </p>
+  ); 
+};
+
+import { useLazyLoadQuery } from "react-relay";
+
+function func2({ query }) {
+  const data = useLazyLoadQuery(query, {}); // $ Source
+  return <p dangerouslySetInnerHTML={{ __html: data.comments[0].text }} />; // $ Alert
+}
+
+import { useQueryLoader, usePreloadedQuery } from "react-relay";
+
+function func3({ initialQueryRef, query }) {
+  const [queryReference, loadQuery] = useQueryLoader(query, initialQueryRef);
+  return (
+    <h1
+      dangerouslySetInnerHTML={{
+        __html: usePreloadedQuery(query, queryReference).user?.name, // $ Alert
+      }}
+    />
+  ); 
+}
+
+import { useClientQuery } from "react-relay";
+
+function func4({ query }) {
+  const data = useClientQuery(query, {}); // $ Source
+  return <h1 dangerouslySetInnerHTML={{ __html: data }} />; // $ Alert
+}
+
+import { useRefetchableFragment } from "react-relay";
+
+function func5({ query, props }) {
+  const [data, refetch] = useRefetchableFragment(query, props.comment); // $ Source
+  return (
+    <>
+      <h1 dangerouslySetInnerHTML={{ __html: data }} /> // $ Alert
```

Comment:
Maybe add a line here to ensure `refetch` isn't seen as a source.
```suggestion
      <h1 dangerouslySetInnerHTML={{ __html: data }} /> // $ Alert
      <h1 dangerouslySetInnerHTML={{ __html: refetch }} /> // OK
```

Reply:
I have added it without `// OK` comment, since I believe we don't write it anymore unless we have extra comment? 
```
<h1 dangerouslySetInnerHTML={{ __html: refetch }} />
```

Reply:
Thanks. The `OK` comment is not important.